pkg/util/validation/limits.go

```diff
@@ -103,6 +104,7 @@ func (l *Limits) RegisterFlags(f *flag.FlagSet) {
 	f.DurationVar(&l.MaxQueryLength, "store.max-query-length", 0, "Limit to length of chunk store queries, 0 to disable.")
 	f.IntVar(&l.MaxQueryParallelism, "querier.max-query-parallelism", 14, "Maximum number of queries will be scheduled in parallel by the frontend.")
 	f.IntVar(&l.CardinalityLimit, "store.cardinality-limit", 1e5, "Cardinality limit for index queries.")
+	f.DurationVar(&l.MaxCacheFreshness, "frontend.per-user-max-cache-freshness", 0, "Most recent allowed cacheable result per-tenant, to prevent caching very recent results that might still be in flux.")
```

Comment:
I think this CLI flag should replace the existing one `frontend.max-cache-freshness`, instead of introducing a new one (at this, this is how all other flags in this struct work). What do you think?

Reply:
That would involve a breaking change to the config file, as the flag would now be a part of the `Limits` struct. If I understand correctly, we should support the old flag for a few releases (I can add a "deprecated" in the flag description).

Reply:
Moving the flag (without renaming it) doesn't introduce any breaking change to the flag itself, but would introduce a breaking change in the YAML config. To overcome the breaking change in the YAML config, you can keep the old config in YAML (without registering its CLI flag), mark it as deprecated in the CHANGELOG and add a `// TODO` comment to remove it in Cortex 1.4.0.